mlrun/serving/states.py

```diff
@@ -733,6 +736,11 @@ def add_route(
         if not route:
             route = TaskStep(class_name, class_args, handler=handler)
         route.function = function or route.function
+
+        if len(self._routes) >= MAX_ALLOWED_NUM_OF_STEPS:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"The maximum allowed number of steps is {MAX_ALLOWED_NUM_OF_STEPS}"
```

Comment:
A bit better sounding:
```suggestion
                f"The maximum number of steps allowed in the serving graph is {MAX_ALLOWED_NUM_OF_STEPS}"
```